src/Orchard/Validation/PathValidation.cs

```diff
@@ -26,10 +31,22 @@ public static class PathValidation {
             }
 
             if (!valid) {
-                throw new ArgumentException("Invalid path");
+                if (System.Environment.OSVersion.Platform == PlatformID.Win32NT)
+                    throw new InvalidWindowsPathException("Invalid path");
+                else
+                    throw new ArgumentException("Invalid path");
             }
 
             return mappedPath;
         }
     }
+
+    /// <summary>
+    /// Thrown when an invalid path is encountered on a Windows platform.
+    /// </summary>
+    [Serializable]
```

Comment:
Not necessary as we can use `OrchardException`

Reply:
Throwing OrchardException breaks unit testOrchard.Framework.Tests.Orchard.Tests.Storage.FileStystemStorageProviderTests.CreateFileFailsInInvalidPath.